manifests/charts/gateways/istio-ingress/templates/deployment.yaml

```diff
@@ -210,6 +210,10 @@ spec:
           - name: {{ $key }}
             value: {{ $val }}
           {{- end }}
+          {{- range $key, $value := .Values.meshConfig.defaultConfig.proxyMetadata }}
```

Comment:
Yes, in particular this was not correct - 'defaultConfig' is different from the 'actual config' for this deployment, i.e. the default is supposed to only provide defaults.

Reply:
Another option I thought of was reading proxy metadata within istio-agent and then dynamically setting the environment. But I think many of our environment variables need to be set when the process starts so not sure its any better